server/ml/priority_queue.go

```diff
@@ -0,0 +1,57 @@
+package ml
+
+import (
+	"container/heap"
+)
+
+// An Item is something we manage in a max heap.
+type Item struct {
+	value    interface{}
+	priority float64
+}
+
+// A MaxHeapK implements heap.Interface and holds k Items with highest priority
```

Comment:
Nit: `k` is not public and hence not important
```suggestion
// A MaxHeapK implements heap.Interface.
```